src/make-should-retry.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+import type {Response as SuperAgentResponse, CallbackHandler} from "superagent";
```

Comment:
These are all `any` because we don't have superagent flow types. I hope to rectify that at some point. For now, this at least helps annotate the code more clearly.